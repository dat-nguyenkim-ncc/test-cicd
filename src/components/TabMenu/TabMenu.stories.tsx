import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import TabMenu, { TabMenuProps } from './TabMenu'
import { BrowserRouter } from 'react-router-dom'
import { Box } from 'theme-ui'

export default {
  title: 'Primitives/TabMenu',
  component: TabMenu,
  parameters: {
    layout: 'padded',
  },
} as Meta

const Template: Story<TabMenuProps> = args => (
  <BrowserRouter>
    <Box sx={{ width: '100%' }}>
      <TabMenu {...args} />
    </Box>
  </BrowserRouter>
)

export const Default = Template.bind({})
Default.args = {
  buttons: [
    { active: true, label: 'Test 1', to: '/' },
    { label: 'Test 2', to: '/' },
    { label: 'Test 2', to: '/' },
    { label: 'Test 2', to: '/' },
    { label: 'Test 2', to: '/' },
    { label: 'Test 2', to: '/' },
  ],
} as TabMenuProps
