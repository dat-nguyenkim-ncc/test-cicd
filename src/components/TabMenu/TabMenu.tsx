import React from 'react'
import { Flex } from 'theme-ui'
import { ViewInterface } from '../../types'
import Link from '../Link'

export type TabMenuProps = ViewInterface<{
  buttons: { active?: boolean; label: string; to: string; disabled?: boolean }[]
}>

const TabMenu = ({ buttons, sx }: TabMenuProps) => {
  return (
    <Flex sx={sx}>
      {buttons.map((b, index) => (
        <Link
          key={index}
          sx={{
            width: `calc(${100 / buttons.length}% + 10px)`,
            ml: index > 0 ? '-10px' : 0,
            zIndex: b.active ? 10 : 0,
            alignItems: 'center',
            display: 'flex',
            justifyContent: 'center',
          }}
          variant={b.active ? 'tabActive' : 'tab'}
          to={b.disabled ? '#' : b.to}
        >
          {b.label}
        </Link>
      ))}
    </Flex>
  )
}

export default TabMenu
