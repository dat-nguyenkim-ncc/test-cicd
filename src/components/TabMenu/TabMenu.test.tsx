import * as React from 'react'
import { shallow } from 'enzyme'
import TabMenu from './TabMenu'

describe('TabMenu', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(
      <TabMenu
        buttons={[
          { active: true, label: 'Test 1', to: '/' },
          { label: 'Test 2', to: '/' },
        ]}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
