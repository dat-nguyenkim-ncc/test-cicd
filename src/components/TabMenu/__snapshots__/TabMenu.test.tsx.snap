// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabMenu matches snapshot 1`] = `ShallowWrapper {}`;
